From 37813b8747dc9633f8cccba4a5a8b86955184789 Mon Sep 17 00:00:00 2001
From: Chris White <cxwembedded@gmail.com>
Date: Mon, 22 Jun 2020 08:28:29 -0400
Subject: [PATCH] Retrieve version when in a git worktree

- Change git-version-gen invocation to match tag format
- Use autoconf to make vala/valaversion.vala rather than generating that
  file by hand
---
 vala/Makefile.am | 12 ++----------
 3 files changed, 8 insertions(+), 15 deletions(-)

diff --git a/vala/Makefile.am b/vala/Makefile.am
index 0191fc9aa..af48bbf0c 100644
--- a/vala/Makefile.am
+++ b/vala/Makefile.am
@@ -11,15 +11,7 @@ AM_CPPFLAGS = \
 	-DPACKAGE_DATADIR=\"$(pkgdatadir)\" \
 	$(NULL)
 
-BUILT_SOURCES = vala.vala.stamp $(srcdir)/valaversion.vala
-
-$(srcdir)/valaversion.vala:
-	sed -e "s#\@VALA_MAJOR_VERSION\@#$(VALA_MAJOR_VERSION)#g" \
-		-e "s#\@VALA_MINOR_VERSION\@#$(VALA_MINOR_VERSION)#g" \
-		-e "s#\@VALA_MICRO_VERSION\@#$(VALA_MICRO_VERSION)#g" \
-		-e "s#\@API_VERSION\@#$(API_VERSION)#g" \
-		-e "s#\@PACKAGE_VERSION\@#$(PACKAGE_VERSION)#g" \
-		< $@.in > $@
+BUILT_SOURCES = vala.vala.stamp
 
 lib_LTLIBRARIES = \
 	libvala@PACKAGE_SUFFIX@.la \
@@ -229,7 +221,7 @@ dist_vapi_DATA = libvala@PACKAGE_SUFFIX@.vapi
 libvala@PACKAGE_SUFFIX@.vapi: $(top_srcdir)/gee/gee.vapi $(top_srcdir)/vala/vala.vapi
 	cat $^ > $@
 
-EXTRA_DIST = $(libvala_la_VALASOURCES) vala.vapi vala.vala.stamp vala.h valaversion.vala.in
+EXTRA_DIST = $(libvala_la_VALASOURCES) vala.vapi vala.vala.stamp vala.h
 
 MAINTAINERCLEANFILES = \
 	vala.vapi \
-- 
GitLab

